src/components/shared/Menu/index.jsx

```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
+import { OuterSpace, StyledMenu } from './menu.styles';
+
+/**
+ * Renders Anchored Menu.
+ *
+ * @param {Object} anchorRef anchor element reference.
+ * @param {boolean} isOpen state of the menu.
+ * @param {children} children JSX.Element.
+ * @param {Function} onClose onClose handler.
+ * @param {boolean} [sticky=false] is anchor sticky.
+ *
+ * @return {JSX.Element}
+ */
+function Menu({ anchorRef, isOpen, children, onClose, sticky }) {
+  const { current } = anchorRef;
+  const [, setReRender] = useState(0);
+
+  const triggerReRender = () => setReRender((prevState) => prevState + 1);
+
+  useEffect(() => {
+    if (isOpen) window.addEventListener('resize', triggerReRender);
+    return () => window.removeEventListener('resize', triggerReRender);
+  }, [isOpen]);
+
+  return (
+    <>
+      <StyledMenu
+        coords={current?.getBoundingClientRect()}
+        isOpen={isOpen}
+        sticky={sticky}
+      >
+        {children}
+      </StyledMenu>
+      <OuterSpace isOpen={isOpen} onClick={onClose} />
+    </>
+  );
+}
+
+Menu.propTypes = {
+  anchorRef: PropTypes.shape({
+    current: PropTypes.instanceOf(Element),
+  }).isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]).isRequired,
+  isOpen: PropTypes.bool.isRequired,
+  onClose: PropTypes.func.isRequired,
+  sticky: PropTypes.bool,
```

Comment:
Boolean naming usually follows a yes/no   
```suggestion
  isSticky: PropTypes.bool,
```

Reply:
done